libs/parsers/src/lib/expression/scanner.ts

```diff
@@ -5,19 +5,22 @@ import { ScannerException } from '../common/exceptions/scanner';
 import { isNumeric, isWhitespace } from '../common/helpers/expression';
 import type { ExpressionToken } from '../models/expression';
 import { ExpressionTokenType } from '../models/expression';
+import type { ConnectionReferenceKeyFormat } from '@microsoft/utils-logic-apps';
 import { equals } from '@microsoft/utils-logic-apps';
 
 export class ExpressionScanner {
   private _expression: string;
   private _startPosition: number;
   private _currentToken: ExpressionToken;
+  private _referenceKeyFormat: ConnectionReferenceKeyFormat | undefined;
```

Comment:
this is a string parser/scanner we cannot introduce product / connector specific thing in this class that way this is no longer a library which can be used outside the context of LA or PA... I am not seeing where you are using this as class property.

Reply:
I'm using the reference in the parser file on line 143. This was introduced at your request to identify if it is openapi:
> You cannot assume here / means it is openApiConnection aliasing format, what if properties actually have this character, so please add some kind of aliasing support flag in the parser so it knows it has to assume / as open api alias parameter name

```       
 if (expression.type === ExpressionType.StringLiteral && scanner.referenceKeyFormat === ConnectionReferenceKeyFormat.OpenApi) {
          // takes care of expressions that are nested such as ['body/value']
          for (const expressionValue of (expression as ExpressionLiteral).value.split('/')) {
```

Reply:
Updated to a generic bool to signify this instead